python/cudaq/kernel/utils.py

```diff
@@ -325,7 +325,7 @@ def mlirTypeFromPyType(argType, ctx, **kwargs):
             return cc.StdvecType.get(ctx, mlirTypeFromPyType(np.complex64, ctx))
 
         if isinstance(argInstance[0], pauli_word):
-            return cc.StdvecType.get(ctx, cc.CharspanType.get(ctx))
+            return cc.StdvecType.get(ctx, mlirTypeFromPyType(pauli_word, ctx))
```

Comment:
Is `mlirTypeFromPyType` only used for arguments?

Reply:
I was asking because some cases have a contextual nuance.

Say we have `list[int]`. Is this type *always* going to be `!stdvec<i32>` in the IR? The answer is no. If it is an argument then we would likely use `!stdvec<i32>` since we don't want to pass ownership to the kernel (necessarily). However, if this is a type of a local variable then `!stdvec<i32>` is wrong since it is a span (reference) over another object. Our local variable is itself an object, so it requires storage. The required type there for the allocation is `!array<i32 x size>`. Of course once we have the storage for the object, we can always wrap it in a span as a `!stdvec<i32>`. Note that a `!charspan<n>` does not have any storage either.